Util/model_parser_cxx/model_parser_data.H

```diff
@@ -0,0 +1,21 @@
+#ifndef MODEL_PARSER_DATA_H
+#define MODEL_PARSER_DATA_H
+
+namespace model
+{
+
+    // integer keys for indexing the model_state array
+
+    constexpr int nvars = 3 + NumSpec;
+    constexpr int idens = 0;
+    constexpr int itemp = 1;
+    constexpr int ipres = 2;
+    constexpr int ispec = 3;
+
+    extern AMREX_GPU_MANAGED int npts;
+    extern AMREX_GPU_MANAGED bool initialized;
+
+    extern amrex::Gpu::ManagedVector<amrex::Real> state;
```

Comment:
If the model data is going to be a std::vector, we cannot directly access it inside a ParallelFor if we're running on the GPU. We'd have to get its `dataPtr()` prior to the ParallelFor and then pass that pointer to the `interpolate` function.

Reply:
I suggest having the data be separate managed Array1Ds (similar to what we do in other places like Helmholtz) for density, pressure, etc., with a fixed size at compile time that is probably bigger than we realistically need. (Even if it's a large number like 10k, it won't be that much memory consumed.) If the user has a larger model than our default, they can change it themselves.

Reply:
but this would mean an Array1D for each species?  So I would store those as a Vector of size NumSpec of Array1D?

Reply:
Since NumSpec is known at compile time, that one can be an Array2D.

Reply:
Ah, indeed.  okay.  I'll do this.  I'll make the number of points a preprocessor variable.

Reply:
okay, I made this all an Array2D, and there is a make variable MAX_NPTS_MODEL that you can set to change the size.  I still have the BC issue with the divu / magvort though.